networking.adoc

```diff
@@ -0,0 +1,1081 @@
+= Networking
+
+toc::[]
+
+
+## Configuring a custom PKI
+
+[source,terminal]
+----
+$ oc create -f user-ca-bundle.yaml
+----
+
+
+[source,terminal]
+----
+$ oc edit proxy/cluster
+----
+
+## Configuring the node port service range
+
+[source,terminal]
+----
+$ oc patch network.config.openshift.io cluster --type=merge -p \
+  '{
+    "spec":
+      { "serviceNodePortRange": "30000-<port>" }
+  }'
+----
+
+## Configuring PTP hardware
+
+[source,terminal]
+----
+$ oc create -f <filename> 
+----
+
+## Configuring ExternalIPs for services
+
+[source,terminal]
+----
+$ oc edit networks.config cluster
+----
+
+## Configuring ingress cluster traffic on AWS using a Network Load Balancer
+
+[source,terminal]
+----
+$ oc create -f ingresscontroller-aws-nlb.yaml
+----
+
+## Configuring ingress cluster traffic using an Ingress Controller
+
+[source,terminal]
+----
+$ oc new-project <project_name>
+----
+
+
+[source,terminal]
+----
+$ oc new-project myproject
+----
+
+
+[source,terminal]
+----
+$ oc new-app \
+    -e MYSQL_USER=admin \
+    -e MYSQL_PASSWORD=redhat \
+    -e MYSQL_DATABASE=mysqldb \
+    registry.redhat.io/rhscl/mysql-80-rhel7
+----
+
+
+[source,terminal]
+----
+$ oc expose service <service_name>
+----
+
+
+[source,terminal]
+----
+$ oc expose service mysql-80-rhel7
+----
+
+## Configuring ingress cluster traffic using a load balancer
+
+[source,terminal]
+----
+$ oc new-project <project_name>
+----
+
+
+[source,terminal]
+----
+$ oc new-project myproject
+----
+
+
+[source,terminal]
+----
+$ oc new-app \
+    -e MYSQL_USER=admin \
+    -e MYSQL_PASSWORD=redhat \
+    -e MYSQL_DATABASE=mysqldb \
+    registry.redhat.io/rhscl/mysql-80-rhel7
+----
+
+
+[source,terminal]
+----
+$ oc expose service <service_name>
+----
+
+
+[source,terminal]
+----
+$ oc expose service mysql-80-rhel7
+----
+
+
+[source,terminal]
+----
+$ oc create -f <file-name>
+----
+
+
+[source,terminal]
+----
+$ oc create -f mysql-lb.yaml
+----
+
+## Configuring ingress cluster traffic using a NodePort
+
+[source,terminal]
+----
+$ oc new-project <project_name>
+----
+
+
+[source,terminal]
+----
+$ oc new-project myproject
+----
+
+
+[source,terminal]
+----
+$ oc new-app \
+    -e MYSQL_USER=admin \
+    -e MYSQL_PASSWORD=redhat \
+    -e MYSQL_DATABASE=mysqldb \
+    registry.redhat.io/rhscl/mysql-80-rhel7
+----
+
+
+[source,terminal]
+----
+$ oc expose dc mysql-80-rhel7 --type=NodePort --name=mysql-ingress
+----
+
+
+[source,terminal]
+----
+$ oc delete svc mysql-80-rhel7
+----
+
+## Configuring ingress cluster traffic for a service external IP
+
+[source,terminal]
+----
+$ oc patch svc <name> -p \
+  '{
+    "spec": {
+      "externalIPs": [ "<ip_address>" ]
+    }
+  }'
+----
```

Comment:
Requires a declarative alternative

Reply:
There is a declarative example right above this patch in the documentation.

Reply:
The concern is that it explicitly says for updating existing services use `oc patch`. Could we perhaps change that wording so user understands they can update services as wel?